ts-framework/functions/src/build/mcp.ts

```diff
@@ -13,13 +13,21 @@ export async function buildMCPServer(options: {
   serverExport: string;
   functionEntrypoint: string;
   outDir: string;
+  envSchema?: {
+    [key: string]: { description?: string };
+  };
 }): Promise<BuildMCPServerResult> {
   await mkdir(options.outDir, { recursive: true });
 
-  const manifest = await buildFunctionsManifest({
+  let manifest = await buildFunctionsManifest({
     entrypoint: options.serverEntrypoint,
     serverExport: options.serverExport,
   });
+
+  if (options.envSchema) {
+    manifest = { ...manifest, variables: options.envSchema };
+  }
```

Comment:
variables are defined at the tool-level in the manifest so you'll need to inject them into each `manifest.tools.*`